dds/idl/typeobject_generator.cpp

```diff
@@ -1732,8 +1732,7 @@ typeobject_generator::get_minimal_type_identifier(AST_Type* type)
   }
 
   HashTypeIdentifierMap::const_iterator pos = hash_type_identifier_map_.find(type);
-  OPENDDS_ASSERT(pos != hash_type_identifier_map_.end());
-  return pos->second.minimal;
+  return pos == hash_type_identifier_map_.end() ? OpenDDS::XTypes::TypeIdentifier() : pos->second.minimal;
```

Comment:
Is there a concern that callers expect a non-default TypeIdentifier?

Reply:
When run with an unsupported type it hit the assert.  It's probably worth a look to see if there would be any issues with supported types.

Reply:
I reviewed the calling code and didn't see any issues.  It may not generate useful typeobjects but those are cases where opendds_idl exits with an error anyway.